src/Cmd.mli

```diff
@@ -1,21 +1,27 @@
 (** Interface for running system commands *)
 
-type t
+open Import
 
-val make :
-  ?env:string Js.Dict.t -> cmd:Path.t -> unit -> (t, string) result Promise.t
+type shell = [ `Shell of string ]
+
+type spawn = [ `Spawn of Path.t * string list ]
```

Comment:
How about we just create a spawn type `{ prog : Path.t ; args : string list }` and then use that instead of the current `spawn`. I'm not a fan of polymorphic variants and try to avoid them as much as possible.